docs/baseband/tutorials/getting_started.rst

```diff
@@ -0,0 +1,317 @@
+the `baseband.data` module)::
+
+    >>> import baseband.vdif as vdif
+    >>> from baseband.data import SAMPLE_VDIF
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs')
+
+To close the file::
+
+    >>> fh.close()
+
+The same syntax can be used to open a file of any format.  To open Baseband's
+sample DADA file, for example::
+
+    >>> from baseband import dada
+    >>> from baseband.data import SAMPLE_DADA
+    >>> fh_dada = dada.open(SAMPLE_DADA, 'rs')
+    >>> dada_data = fh_dada.read()
+    >>> fh_dada.close()
+
+In general, file I/O and data manipulation use the same syntax across all
+file formats.  When using ``open`` for Mark 4 and Mark 5B files, however, two
+keywords - ``ntrack``, and ``decade`` - may need to be set manually.  For these
+and VDIF, ``frames_per_second`` may also need to be passed if it can't be read
+or inferred from the file. Notes on such features and quirks of individual
+formats can be found within the :ref:`Specific file format
+<specific_file_formats_toc>` section.
+
+For now, let's go back to using VDIF files.
+
+Decoding Data and the Sample File Pointer
+-----------------------------------------
+
+We gave `~baseband.vdif.open` the ``'rs'`` flag, which opens the file in 
+"stream reader" mode.  The function returns an instance of
+`~baseband.vdif.base.VDIFStreamReader`, a wrapper around `io.BufferedReader`
+that adds methods to decode files as data frames and seek to and read individual
+data samples.  To read the first 12 data samples into an `~numpy.ndarray`, we
+would use the `~baseband.vdif.base.VDIFStreamReader.read` method::
+
+    >>> fh = vdif.open(SAMPLE_VDIF, 'rs')
+    >>> d = fh.read(12)
+    >>> type(d)
+    <... 'numpy.ndarray'>
+    >>> d.shape
+    (12, 8)
+    >>> d[:, 0].astype(int)  # First thread.
+    array([-1, -1,  3, -1,  1, -1,  3, -1,  1,  3, -1,  1])
+
+The sample VDIF file has 8 concurrent frequency bands, or "channels", which are
+mapped as the columns of the array.
+
+We can access information from the header by printing ``fh``::
+
+    >>> fh
+    <VDIFStreamReader name=... offset=12
+        nthread=8, samples_per_frame=20000, nchan=1,
+        frames_per_second=1600, complex_data=False, bps=2, edv=3,
+        station=65532, (start) time=2014-06-16T05:56:07.000000000>
+
+The ``offset`` gives the current location of the sample file pointer - it's at
+``12`` since we have just read in 12 samples.  If we called ``fh.read(12)``
+again we would get the next 12 samples.  If we instead called  ``fh.read()``, it
+would read from the pointer's *current* position to the end of the file.  If we
+wanted all the data in one array, we would move the file pointer back to the
+start of file, using ``fh.seek``, before reading::
+
+    >>> fh.seek(0)      # Seek to sample 0.  Seek returns its offset in counts.
+    0
+    >>> d_complete = fh.read()
+    >>> d_complete.shape
+    (40000, 8)
+
+.. note::
+
+    ``fh.read()`` returns a **copy** of the data from ``fh``.
```

Comment:
Maybe "return **decoded** data from ``fh``. For typical files, the resulting arrays are far too large to hold in memory."

Reply:
I was originally hoping to point out that, given `d = fh.read()`, any changes made to `d` don't propagate back to `fh`, but on second though few users will care about this (since there's a separate protocol for writing).  I've now changed it, per the suggestion, to warn users from decoding large blocks of data.